### ISSUE TITLE
[Feature] Handle with not existing keys

### ISSUE THREAD

**Issue by xxxcoltxxx** [hairyhenderson/gomplate#1948]
Hello!
It would be nice to handle non-existing keys.
It will provide flexible control when working with non-existent keys and will make the template more readable.
It allows as to control where we should fallback to some default value and where we should fail template rendering.
Its easy to implement - just add command-line flag that controls the `missingkey=error|zero` option.

Current behavior:
```bash
$ gomplate -i 'Hi {{ .name }}'
Hi 14:06:57 ERR  error="failed to render template <arg>: template: <arg>:1:6: executing \"<arg>\" at <.name>: map has no entry for key \"name\""
$ gomplate -i 'Hi {{ if (has . "name") }}{{ .name }}{{ else }}Alex{{end}}'
Hi Alex
```

Proposed behavior:

```bash
$ gomplate --missing-key error -i 'Hi {{ .name }}'
Hi 14:06:57 ERR  error="failed to render template <arg>: template: <arg>:1:6: executing \"<arg>\" at <.name>: map has no entry for key \"name\""

$ gomplate --missing-key zero -i 'Hi {{ .name }}'
Hi <no value>

$ gomplate --missing-key zero -i 'Hi {{ .name | default "Alex" }}'
Hi Alex

$ gomplate --missingkey zero -i 'Hi {{ .name | required }}'
Hi 14:12:04 ERR  error="failed to render template <arg>: template: <arg>:1:11: executing \"<arg>\" at <required>: error calling required: can not render template: a required value was not set"
```

**Comment by hairyhenderson** [comment]
Hi @xxxcoltxxx, thanks for filing this and thanks for the PR.

This has come up before, in #310 and #1362 at least, and I think in other issues as well.

I'm not sure why a flag didn't occur to me for controlling it, but at this point it seems reasonable.

**Comment by gjrtimmer** [comment]
Is there any update on this? I hope this one makes it to a release soon, kinda need it.

**Comment by hairyhenderson** [comment]
@gjrtimmer thanks for prodding - the PR is stuck, I will unstick it.

As for a release, this will be in v4, which won't be "soon" (depending on what "soon" means to you). In the meantime, there are workarounds (the [`test`](https://docs.gomplate.ca/functions/test/) namespace and [`coll.Has`](https://docs.gomplate.ca/functions/coll/#collhas)).

**Comment by hairyhenderson** [comment]
Fixed by #1949
